src/deepMerge.test.ts

```diff
@@ -0,0 +1,50 @@
+import { deepMerge } from './deepMerge';
+import { expect } from 'vitest';
+
+describe('deepMerge', () => {
+  test('should handle undefined inputs', () => {
+    expect(deepMerge(undefined, undefined)).toBeUndefined();
+    expect(deepMerge(undefined, 5)).toBe(5);
+    expect(deepMerge(5, undefined)).toBe(5);
+  });
+
+  test('should handle null inputs', () => {
+    expect(deepMerge(null, null)).toBeNull();
+    expect(deepMerge(null, 5)).toBe(5);
+    expect(deepMerge(5, null)).toBe(5);
+  });
+
+  test('should merge non-object and non-array primitives', () => {
+    expect(deepMerge('foo', 'bar')).toBe('foo');
+    expect(deepMerge('foo', undefined)).toBe('foo');
+    expect(deepMerge(undefined, 'bar')).toBe('bar');
+  });
+
+  test('should merge objects', () => {
+    const a = { foo: 'bar', x: 1 };
+    const b = { foo: 'baz', y: 2 };
+    expect(deepMerge(a, b)).toEqual({ foo: 'bar', x: 1, y: 2 });
+  });
```

Comment:
It seems that all tests should pass the usual "merge". Maybe add tests with nested objects?

Reply:
Added case for nested objects